osbs/http.py

```diff
@@ -59,8 +82,24 @@ def request(self, url, *args, **kwargs):
             with stream as s:
                 content = s.req.content
                 return HttpResponse(s.status_code, s.headers, content)
-        except requests.exceptions.HTTPError as ex:
-            raise OsbsNetworkException(url, str(ex), ex.response.status_code,
+        except (HTTPError, RetryError) as ex:
+            if isinstance(ex, RetryError):
+                # RetryError have neither response nor status code set
+                reason = ex.args[0].reason
+                # Try to match it, as ancient python-requests in rhel is ancient
+                if isinstance(reason, ResponseError):
+                    if requests.__version__.startswith('2.6.'):
+                        regexp = r"too many (\d+) error responses"
+                        match = re.search(regexp, reason.message)
+                        if match and len(match.group()) > 0:
+                            status_code = match.group(1)
```

Comment:
IIRC at this point some parts are comparing status code in `OsbsNetworkException` - this should go away.

I think in case of retries it makes sense to avoid status code there - it may throw different codes in various attempts